### PR TITLE
fix: styles when clicking the CodeInput and this have an error

### PR DESCRIPTION
**Pull request by blakmetall**
fix: #1551 

## Changes proposed in this PR:
- fix: styles when clicking the CodeInput and this have an error

- [ X ] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

File: src/components/CodeInput/styled/index.js [hunk 1]
```diff
@@ -77,8 +77,12 @@ const StyledInput = attachThemeAttrs(styled.input)`
         :focus,
         :active {
             box-shadow: none;
-            background-color: ${props.palette.background.main};
-            border: 1px solid ${props.palette.border.main};
+
+            ${!props.error &&
```
Comment (TahimiLeonBravo) [inline src/components/CodeInput/styled/index.js:81]: Since using `!` in conditions is more difficult to read, we should avoid using it on conditions.

Comment (LeandroTorresSicilia) [reply src/components/CodeInput/styled/index.js:81]: it makes the code more complex and difficult for developer's understanding

Comment (blakmetall) [reply src/components/CodeInput/styled/index.js:81]: sure

File: src/components/CodeInput/styled/index.js [hunk 2]
```diff
@@ -77,8 +81,12 @@ const StyledInput = attachThemeAttrs(styled.input)`
...
         :active {
             box-shadow: none;
-            background-color: ${props.palette.background.main};
-            border: 1px solid ${props.palette.border.main};
+
+            ${props.renderActiveInputStyles &&
+                `
+                background-color: ${props.palette.background.main};
+                border: 1px solid ${props.palette.border.main};
+            `}
```
Comment (yvmunayev) [inline src/components/CodeInput/styled/index.js:89]: @blakmetall I think if you put the isActive block before the Error block, the problem is solved

Comment (blakmetall) [reply src/components/CodeInput/styled/index.js]: sure, I improved now based on your hint

Comment (blakmetall) [reply src/components/CodeInput/styled/index.js]: sure, logic improved based on your hint

File: src/components/CodeInput/styled/index.js [hunk 3]
```diff
@@ -60,26 +69,6 @@ const StyledInput = attachThemeAttrs(styled.input)`
         background-color: ${props.palette.background.main};
         border: 2px solid ${props.palette.error.main};
         background-clip: padding-box;
-
-        :focus,
-        :active {
-            background-color: ${props.palette.background.main};
-            border: 2px solid ${props.palette.error.main};
-            box-shadow: ${props.shadows.error};
```
Comment (yvmunayev) [inline src/components/CodeInput/styled/index.js:68]: here you can not remove the shadow when doing focus

Comment (blakmetall) [reply src/components/CodeInput/styled/index.js:68]: sorry, my bad... now is returned that specific block, the "box-shadow" along with a few properties that were already there.

**Pull request merged**